pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -13,25 +31,73 @@ import (
 	"vitess.io/vitess/go/vt/wrangler"
 )
 
-func (r *ReconcileVitessKeyspace) reconcileResharding(ctx context.Context, vtk *planetscalev2.VitessKeyspace) error {
-	wr, ts, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
+const (
+	maxSafeVReplicationLag = 10
+)
+
+func (r *reconcileHandler) reconcileResharding(ctx context.Context) error {
+	wr, ts, err := newWrangler(ctx, r.vtk.Spec.GlobalLockserver)
 	if err != nil {
 		return err
 	}
 	defer ts.Close()
 
-	workflows, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
+	workflowList, err := wr.ListAllWorkflows(ctx, r.vtk.Spec.Name)
 	if err != nil {
-		workflows = make([]string, 0)
+		workflowList = make([]string, 0)
 	}
 
-	vtk.Status.ActiveWorkflows = workflows
-	// TODO: Address when ConditionUnknown is applicable.
-	if len(workflows) != 0 {
-		vtk.Status.ReshardingInProgress = corev1.ConditionTrue
-	} else {
-		vtk.Status.ReshardingInProgress = corev1.ConditionFalse
+	reshardingInProgress := corev1.ConditionUnknown
+	workflows := make([]planetscalev2.WorkflowStatus, 0, len(workflowList))
+	for _, workflowName := range workflowList {
+		workflow, err := wr.ShowWorkflow(ctx, workflowName, r.vtk.Spec.Name)
+		if err != nil {
+			return err
+		}
+		workflowStatus := planetscalev2.WorkflowStatus{
+			Workflow: workflow.Workflow,
+		}
+
+		// We aggregate status across all the shards for the workflow so we can definitely know if we are in two states:
+		// Copying, or Error. We also do this so we can determine what all of the serving shards are.
+		// At a high level we mostly need to know if we are still in the Copying phase (for any shard whatsoever), or if
+		// we have an error in resharding somewhere that needs to be surfaced.
+		for name, status := range workflow.ShardStatuses {
+			if status.MasterIsServing {
+				shard := strings.Split(name, "/")[0]
+				r.vtk.Status.ServingShards = append(r.vtk.Status.ServingShards, shard)
+			}
+			for _, vReplRow := range status.MasterReplicationStatuses {
+				if vReplRow.State == "Error" {
+					workflowStatus.State = planetscalev2.ErrorState
+					break
+				}
+				if vReplRow.State == "Copying" {
+					workflowStatus.State = planetscalev2.CopyingState
+				}
+				if vReplRow.State == "Lagging" && workflowStatus.State != planetscalev2.CopyingState {
+					workflowStatus.State = planetscalev2.LaggingState
+				}
+				if vReplRow.State == "Running" && workflowStatus.State == "" {
+					workflowStatus.State = planetscalev2.RunningState
+				}
+			}
+		}
+		workflows = append(workflows, workflowStatus)
+
+		if workflow.SourceLocation.Keyspace == workflow.TargetLocation.Keyspace {
+			reshardingInProgress = corev1.ConditionTrue
+		} else if reshardingInProgress != corev1.ConditionTrue {
+			reshardingInProgress = corev1.ConditionFalse
+		}
+
+		// If MaxVReplicationLag ever exceeds max safe value, we need to set UnsafeVReplicationLag to true.
```

Comment:
+1 to @enisoc's comment about making the variable's `true` state the positive one, so the logic would change here.